### PR TITLE
chore(repo): Only allow snapshot releases for Clerk org members

### PR DESCRIPTION
**Pull request by nikosdouvlis**
## Description
This limits the `!snapshot` command to Clerk org members only, preventing malicious 3rd-party users from opening a PR and releasing a `snapshot` test version with unverified code.

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerkinc/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [x] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

### CODE REVIEW

File: .github/workflows/release-snapshot.yml [hunk 1]
```diff
@@ -17,6 +17,22 @@ jobs:
       TURBO_TOKEN: ${{ secrets.TURBO_TOKEN }}
       TURBO_TEAM: ${{ vars.TURBO_TEAM }}
     steps:
+      - name: Limit action for Clerk members
```
Comment (LekoArts) [inline .github/workflows/release-snapshot.yml:20]:
```suggestion
      - name: Limit action to Clerk members
```

Nit

Comment (nikosdouvlis) [reply .github/workflows/release-snapshot.yml]: Thank you, fixed

**Review by LekoArts — APPROVED** [review]

**Review by dimkl — APPROVED** [review]

**Review by anagstef — APPROVED** [review]
💯

**Review by SokratisVidros — APPROVED** [review]

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**